src/components/structures/MessagePanel.js

```diff
@@ -430,11 +430,11 @@ module.exports = React.createClass({
         // as 'today' for the date separators.
```

Comment:
sorry, I should have been clearer.

The comment isn't just referring to the `DateSeparator` itself, but also the `_wantsDateSeparator` calculation. Consider the behaviour when:
 * a message was sent before midnight, but we haven't yet received the remote echo for it
 * we received other messages after midnight.

Local echoes are always shown at the end of the timeline so we're going to see a date separator between today's received messages and the local echoes.

> The ts1 given to the DateSeperator etc. is a number of millis, not a date instance.

and that might be a thing we should fix, to save it turning it back into another Date, but it's a separate issue.